app/src/main/java/com/codingblocks/cbonlineapp/fragments/MyCoursesFragment.kt

```diff
@@ -51,7 +51,8 @@ class MyCoursesFragment : Fragment(), AnkoLogger {
         courseDataAdapter =
             CourseDataAdapter(
                 ArrayList(),
-                activity!!,
+                viewModel.courseDao,
```

Comment:
why this ?

Reply:
You were passing the activity, that is never a good practise. Check the line below this, it is the best practise to use context in any fragment

Reply:
And the dao. Because we were initializing the dao inside the adapter everytime a view was binding.